File: problems-extend/3901-4000/3959_check-good-integer/solution.py
class Solution:
    def checkGoodInteger(self, n: int) -> bool:
        digit_sum = 0
        square_sum = 0
        while n:
            digit = n % 10
            digit_sum += digit
            square_sum += digit * digit
            n //= 10
        return square_sum - digit_sum >= 50

File: problems-extend/3901-4000/3959_check-good-integer/solution.rs
impl Solution {
    pub fn check_good_integer(mut n: i32) -> bool {
        let mut digit_sum = 0i64;
        let mut square_sum = 0i64;
        while n > 0 {
            let digit = (n % 10) as i64;
            digit_sum += digit;
            square_sum += digit * digit;
            n /= 10;
        }
        square_sum - digit_sum >= 50
    }
}
